plugin/test/unit/helpers/draft.config.spec.ts

```diff
@@ -0,0 +1,198 @@
+        expectedDataFile,
+        expectedFileName,
+      )
+    })
+
+    it('should create expected metadata file in cached directory', async () => {
+      const expectedFileName = `${cacheDir}/dataMetadata.json`
+
+      await createMetadataFileAndCopyDatastore(publishDir, cacheDir)
+
+      expect(mockEnsureFileSync).toHaveBeenCalledWith(expectedFileName)
+    })
+
+    it('should write expected metadata payload in cached directory', async () => {
+      const expectedFileName = `${cacheDir}/dataMetadata.json`
+      const expectedPayload = {
+        fileName: mockedFileName,
+      }
+
+      await createMetadataFileAndCopyDatastore(publishDir, cacheDir)
+
+      expect(mockWriteJSON).toHaveBeenCalledWith(
+        expectedFileName,
+        expectedPayload,
+      )
+    })
+  })
+
+  describe('when previous filename datastore file already exists in published directory', () => {
+    const mockedDatastoreFile = 'data-test-uuid.mdb'
+    const mockFilesInDirectory = ['testFile.mdb', mockedDatastoreFile]
+
+    beforeEach(() => {
+      mockReadJSON.mockResolvedValue(null)
+      mockReaddir.mockResolvedValue(mockFilesInDirectory)
+    })
+
+    it('should create expected filename in published directory', async () => {
+      const expectedFileName = `${publishDir}/${mockedDatastoreFile}`
+      await createMetadataFileAndCopyDatastore(publishDir, cacheDir)
+
+      expect(mockEnsureFileSync).toHaveBeenCalledWith(expectedFileName)
+    })
+
+    it('should copy cached datastore file to published directory', async () => {
+      const expectedDataFile = join(`${cacheDir}/data/datastore/data.mdb`)
+      const expectedFileName = `${publishDir}/${mockedDatastoreFile}`
+
+      await createMetadataFileAndCopyDatastore(publishDir, cacheDir)
+
+      expect(mockCopySync).toHaveBeenCalledWith(
+        expectedDataFile,
+        expectedFileName,
+      )
+    })
+
+    it('should create expected metadata file in cached directory', async () => {
+      const expectedFileName = `${cacheDir}/dataMetadata.json`
+
+      await createMetadataFileAndCopyDatastore(publishDir, cacheDir)
+
+      expect(mockEnsureFileSync).toHaveBeenCalledWith(expectedFileName)
+    })
+
+    it('should write expected metadata payload in cached directory', async () => {
+      const expectedFileName = `${cacheDir}/dataMetadata.json`
+      const expectedPayload = {
+        fileName: mockedDatastoreFile,
+      }
+
+      await createMetadataFileAndCopyDatastore(publishDir, cacheDir)
+
+      expect(mockWriteJSON).toHaveBeenCalledWith(
+        expectedFileName,
+        expectedPayload,
+      )
+    })
+  })
+
+  describe('when previous datastore file does not exist', () => {
+    const mockedUuid = 'fake-uuid'
+    const expectedUuidFileName = `data-${mockedUuid}.mdb`
+
+    beforeEach(() => {
+      mockReadJSON.mockResolvedValue(null)
+      mockReaddir.mockResolvedValue([])
+      mockUuidv4.mockReturnValue(mockedUuid)
```

Comment:
We don't necessarily have to mock out the `uuidv4` call but by doing so we can explicitly control and check its value when added to the filename.